generators/server/index.js

```diff
@@ -150,13 +150,13 @@ module.exports = JhipsterServerGenerator.extend({
             }
 
             // force variables unused by microservice applications
-            if (this.applicationType === 'microservice') {
+            if (this.applicationType == 'microservice' || this.applicationType == 'uaa') {
                 this.clusteredHttpSession = 'no';
                 this.websocket = 'no';
             }
 
             var serverConfigFound = this.packageName !== undefined &&
-                this.authenticationType !== undefined &&
+                (this.authenticationType !== undefined || (this.applicationType !== undefined && this.applicationType == 'gateway') )&&
```

Comment:
without this line you are not able to generate a gateway using a ready .yo-rc.json.